src/Core/Manifest/ModuleManifest.php

```diff
@@ -225,4 +238,31 @@ public function getModules()
     {
         return $this->modules;
     }
+
+    /**
+     * Get modules sorted by priority
+     *
+     * @return Module[]
+     */
+    public function getSortedModules()
```

Comment:
Per our discussion, `getModules()` is necessary as an option that is not `Config` dependent. This is vital during the boot process when `Config` may not be ready.

Moving this to `regenerate()` may circumvent the issue entirely. `getModules()` would return an unsorted list pre-boot and a sorted one post-boot.